etl/harmonize.py

```diff
@@ -0,0 +1,260 @@
+#
+#  harmonize.py
+#  etl
+#
+
+from pathlib import Path
+from typing import DefaultDict, Dict, List, Optional, Set
+from collections import defaultdict
+import json
+import cmd
+
+import click
+import pandas as pd
+from thefuzz import process
+
+from owid.catalog import Table, Dataset
+
+from etl.paths import REFERENCE_DATASET
+
+
+@click.command()
+@click.argument("data_file")
+@click.argument("column")
+@click.argument("output_file")
+def harmonize(data_file: str, column: str, output_file: str) -> None:
+    """
+    Given a data file in feather or CSV format, and the name of the columnn representing
+    country or region names, interactively generate a JSON mapping from the given names
+    to OWID's canonical names.
+
+    When a name is ambiguous, you can use:
+
+    n: to ignore and skip to the next one
+
+    s: to suggest matches; you can also enter a manual match here
+
+    q: to quit and save the partially complete mapping
+
+    If a mapping file already exists, it will resume where the mapping file left off.
+    """
+    t = read_table(data_file)
+    geo_column = t.reset_index()[column]
+
+    if Path(output_file).exists():
+        print("Resuming from existing mapping...")
+        with open(output_file, "r") as istream:
+            mapping = json.load(istream)
+    else:
+        mapping = {}
+
+    mapping = interactive_harmonize(geo_column, mapping)
+    with open(output_file, "w") as ostream:
+        json.dump(mapping, ostream, indent=2)
+
+
+def read_table(input_file: str) -> Table:
+    if input_file.endswith(".feather"):
+        return Table.read_feather(input_file)
+
+    if input_file.endswith(".csv"):
+        return Table.read_csv(input_file)
```

Comment:
```suggestion
    metadata_filename = Path(input_file).with_suffix(".meta.json")
    if input_file.endswith(".feather"):
        if metadata_filename.exists():
            return Table.read_feather(input_file)
        else:
            return pd.read_feather(input_file)

    if input_file.endswith(".csv"):
        if metadata_filename.exists():
            return Table.read_csv(input_file)
        else:
            return pd.read_csv(input_file)
```

Reply:
Ah, good call. Actually, let's just use pandas directly, no need to go via `Table`. It makes it more flexible.